src/main/java/com/google/neighborgood/servlets/MyTaskServlet.java

```diff
@@ -48,40 +53,66 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       response.sendRedirect(userService.createLoginURL("/account.jsp"));
       return;
     }
-    // keyword is either "Owner" or "Helper". Depending on the keyword, the filter will filter
-    // on different fields accordingly.
-    Filter filter =
-        new FilterPredicate(
-            keyword, FilterOperator.EQUAL, userService.getCurrentUser().getUserId());
 
-    // Depending on the input complete parameter, the status filter will be different.
-    String complete = request.getParameter("complete");
-    if (complete.equals("True")) {
-      Filter openFilter = new FilterPredicate("status", FilterOperator.EQUAL, "COMPLETE");
-      Filter inProgressFilter =
-          new FilterPredicate("status", FilterOperator.EQUAL, "COMPLETE: AWAIT VERIFICATION");
-      CompositeFilter statusFilter = CompositeFilterOperator.or(openFilter, inProgressFilter);
-      filter = CompositeFilterOperator.and(filter, statusFilter);
-    } else {
-      Filter openFilter = new FilterPredicate("status", FilterOperator.EQUAL, "OPEN");
-      Filter inProgressFilter = new FilterPredicate("status", FilterOperator.EQUAL, "IN PROGRESS");
-      CompositeFilter statusFilter = CompositeFilterOperator.or(openFilter, inProgressFilter);
-      filter = CompositeFilterOperator.and(filter, statusFilter);
-    }
+    List<FetchOptions> fetchOptions = new ArrayList<>();
+    FetchOptions firstFetchOption = FetchOptions.Builder.withLimit(PAGE_SIZE);
+    FetchOptions secondFetchOption = FetchOptions.Builder.withLimit(PAGE_SIZE);
 
-    Query query =
-        new Query("Task").addSort("timestamp", SortDirection.DESCENDING).setFilter(filter);
+    // If the client requires for a cursor, get the cursor string given
+    String firstStartCursor = request.getParameter("firstcursor");
+    String secondStartCursor = request.getParameter("secondcursor");
+    if (firstStartCursor != null && secondStartCursor != null) {
```

Comment:
Actually I don't think user should be able to pass only one cursor, since the script file will automatically pass the two cursor strings. But I would make a change in the new commit to deal with the case that only one cursor is passed to the servlet file.